### PR TITLE
[client-preset] Preserving Array<T> or ReadonlyArray<T> in useFragment() data

### PR DESCRIPTION
**Pull request by rachel-church**
## Description

Updating the `useFragment()` (or renamed equivalent) type overload to preserve the `readonly` or lack of modifier when an array is provided as the data argument.

Related #[(issue)](https://github.com/dotansimha/graphql-code-generator/issues/9803)

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by n1ru4l — APPROVED** [review]

**Comment by n1ru4l** [comment]
Hey @rachel-church, can you have a look at the failing tests?

**Comment by rachel-church** [comment]
> Hey @rachel-church, can you have a look at the failing tests?

Done!

File: .changeset/proud-cycles-design.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@graphql-codegen/client-preset': minor
+---
+
+Preserving Array<T> or ReadonlyArray<T> in useFragment() return type.
```
Comment (n1ru4l) [inline .changeset/proud-cycles-design.md:5]:
```suggestion
Preserving `Array<T>` or `ReadonlyArray<T>` in `useFragment()` return type.
```

**Comment by n1ru4l** [comment]
@rachel-church Thank you for the contribution!

**Pull request merged**